src/os_linux.cpp

```diff
@@ -430,4 +432,221 @@ bool OS::checkPreloaded() {
     return dl_iterate_phdr(checkPreloadedCallback, (void*)info) == 1;
 }
 
+static inline bool isNum(const char* s) noexcept {
+    uint8_t c;
+    while ((c = *s++)) {
+        if (c < '0' || c > '9') {
+            return false;
+        }
+    }
+    return true;
+}
+
+static inline int parsePid(const char* s) noexcept {
+    int pid = 0;
+    for (uint8_t c; (c = *s++); ) {
+        pid = pid * 10 + (c - '0');
+    }
+    return pid;
+}
+
+void OS::getProcessIds(int* pids, int* count, int max_pids) {
```

Comment:
IMO, API would be a bit nicer if the function returned `count` instead of setting the output parameter.